models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php

```diff
@@ -415,4 +415,71 @@ public function loadLazyBrickField(DataObject\Objectbrick\Data\AbstractData $ite
             }
         }
     }
+
+    /**
+     * @internal trigger deprecation error when a relation is passed multiple times, remove in Pimcore 7
+     *
+     * @param array|null $data
+     * @param DataObject\Concrete|DataObject\Localizedfield|DataObject\Objectbrick\Data\AbstractData|\Pimcore\Model\DataObject\Fieldcollection\Data\AbstractData $object
+     * @param array $params
+     *
+     * @return array
+     *
+     * @throws \Exception
+     */
+    public function filterMultipleAssignments($data, $object, $params)
+    {
+        if(!is_array($data) || count($data) < 2) {
+            return $data;
+        }
+
+        if (!method_exists($this, 'getAllowMultipleAssignments') || !$this->getAllowMultipleAssignments()) {
+                $relationItems = [];
+                $objectId = null;
+                $fieldName = $this->getName();
+
+                if ($object instanceof DataObject\Concrete) {
+                    $objectId = $object->getId();
+                } elseif (
+                    $object instanceof DataObject\Fieldcollection\Data\AbstractData ||
+                    $object instanceof DataObject\Localizedfield ||
+                    $object instanceof DataObject\Objectbrick\Data\AbstractData
+                ) {
+                    $objectFromContainer = $object->getObject();
+                    if ($objectFromContainer) {
+                        $objectId = $objectFromContainer->getId();
+                    }
+                }
+
+                foreach ($data as $item) {
+                    $elementHash = null;
+                    if ($item instanceof DataObject\Data\ObjectMetadata || $item instanceof DataObject\Data\ElementMetadata) {
+                        if ($item->getElement() instanceof Element\ElementInterface) {
+                            $elementHash = Element\Service::getElementHash($item->getElement());
+                        }
+                    } elseif ($item instanceof Element\ElementInterface) {
+                        $elementHash = Element\Service::getElementHash($item);
+                    }
+
+                    if($elementHash === null) {
```

Comment:
please check if this is still possible. on save every target must have an ID anyway

Reply:
yes, it is possible as you can save targets without ID however not sure what could be the use case.